src/sync/condvar.rs

```diff
@@ -38,6 +38,22 @@ impl Condvar {
         Ok(guard)
     }
 
+    /// Blocks the current thread until this condition variable receives a
+    /// notification and the given condition returns false.
+    pub fn wait_while<'a, T, F>(
+        &self,
+        mut guard: MutexGuard<'a, T>,
+        mut condition: F,
+    ) -> LockResult<MutexGuard<'a, T>>
```

Comment:
I wonder if we could elide these?

Reply:
No, I tried. Eliding the lifetime completely gives a warning because it’s part of the MutexGuard struct. Trying to make it anonymous gives a complete error because it tries to use the same lifetime as &self. 

Also messing with signature seems like a bad idea since we’re trying match what’s in std.